config/filter.d/postfix.conf

```diff
@@ -15,6 +15,7 @@ _daemon = postfix/(submission/)?smtp(d|s)
 failregex = ^%(__prefix_line)sNOQUEUE: reject: RCPT from \S+\[<HOST>\]: 554 5\.7\.1 .*$
             ^%(__prefix_line)sNOQUEUE: reject: RCPT from \S+\[<HOST>\]: 450 4\.7\.1 Client host rejected: cannot find your hostname, (\[\S*\]); from=<\S*> to=<\S+> proto=ESMTP helo=<\S*>$
             ^%(__prefix_line)sNOQUEUE: reject: RCPT from \S+\[<HOST>\]: 450 4\.7\.1 : Helo command rejected: Host not found; from=<> to=<> proto=ESMTP helo= *$
+            ^%(__prefix_line)sNOQUEUE: reject: EHLO from \S+\[<HOST>\]: 504 5\.5\.2 <\S+>: Helo command rejected: need fully-qualified hostname;
```

Comment:
With SMTP I assume that if a 'smtp client' does not have a full domain name it can not be a valid server. All smtp agent of people in my organisation are able to use my postfix SMTP relay with valid credentials that cause the SMTP daemon to bypass EHLO verification.
It's that why I think this rule is justified to limit spam.
For example : Three days ago, a malicious agent have made my postfix bad by trying to introduce itself with a bad SMTP ehlo string. And of course that had cause some thousand of postmaster mails.

If you don't want to check the format of the EHLO command you can remove the postfix configuration option from main configuration file. In this case, this new fail2ban regex will be useless and will not impact the performance (or a little).
But for the system's administrators that want to be as 'strict' as me this regexp will prevent some malicious behaviour.